--- pr/pr.c.orig	2021-04-09 02:24:12.000000000 +0200
+++ pr/pr.c	2021-06-12 06:47:02.275085923 +0200
@@ -65,6 +65,7 @@ __FBSDID("$FreeBSD$");
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
+#include <time.h>
 
 #include "pr.h"
 #include "extern.h"
@@ -1547,7 +1548,7 @@ prtail(int cnt, int incomp)
  * terminate():	when a SIGINT is recvd
  */
 void
-terminate(int which_sig __unused)
+terminate(int which_sig __attribute__((unused)))
 {
 	flsh_errs();
 	exit(1);
@@ -1857,7 +1858,9 @@ setup(int argc, char *argv[])
 
 	(void) setlocale(LC_TIME, (Lflag != NULL) ? Lflag : "");
 
-	d_first = (*nl_langinfo(D_MD_ORDER) == 'd');
+	d_first = 0;
+	if (strlen(nl_langinfo(D_FMT)) >= 2 && nl_langinfo(D_FMT)[1] == 'd')
+		d_first = 1;
 	timefrmt = strdup(d_first ? TIMEFMTD : TIMEFMTM);
 
 	return(0);
