--- logname/logname.c.orig	2021-04-09 02:24:12.000000000 +0200
+++ logname/logname.c	2021-06-12 06:47:02.271086318 +0200
@@ -41,7 +41,6 @@ static const char sccsid[] = "@(#)lognam
 #include <sys/cdefs.h>
 __FBSDID("$FreeBSD$");
 
-#include <capsicum_helpers.h>
 #include <err.h>
 #include <unistd.h>
 #include <stdio.h>
@@ -50,13 +49,10 @@ __FBSDID("$FreeBSD$");
 void usage(void);
 
 int
-main(int argc, char *argv[] __unused)
+main(int argc, char *argv[] __attribute__((unused)))
 {
 	char *p;
 
-	if (caph_limit_stdio() < 0 || caph_enter() < 0)
-		err(1, "capsicum");
-
 	if (argc != 1)
 		usage();
 	if ((p = getlogin()) == NULL)
