--- split.c.orig	2020-06-27 17:37:25.930824801 -0400
+++ split.c	2020-06-29 12:36:51.203153850 -0400
@@ -43,6 +43,8 @@
 #include <unistd.h>
 #include <regex.h>
 
+#include "compat.h"
+
 #define DEFLINE	1000			/* Default num lines per file. */
 
 ssize_t	 bytecnt;			/* Byte count to split on. */
@@ -67,9 +69,6 @@
 	char *ep, *p;
 	const char *errstr;
 
-	if (pledge("stdio rpath wpath cpath", NULL) == -1)
-		err(1, "pledge");
-
 	while ((ch = getopt(argc, argv, "0123456789a:b:l:p:-")) != -1)
 		switch (ch) {
 		case '0': case '1': case '2': case '3': case '4':
