--- tee.c.orig	2020-06-25 10:25:58.600582888 -0400
+++ tee.c	2020-06-26 10:15:52.646544752 -0400
@@ -72,9 +72,6 @@
 	int append, ch, exitval;
 	char buf[8192];
 
-	if (pledge("stdio wpath cpath", NULL) == -1)
-		err(1, "pledge");
-
 	SLIST_INIT(&head);
 
 	append = 0;
@@ -107,9 +104,6 @@
 		argv++;
 	}
 
-	if (pledge("stdio", NULL) == -1)
-		err(1, "pledge");
-
 	while ((rval = read(STDIN_FILENO, buf, sizeof(buf))) > 0) {
 		SLIST_FOREACH(p, &head, next) {
 			n = rval;
