--- ls/cmp.c.orig	2021-04-09 02:24:01.000000000 +0200
+++ ls/cmp.c	2021-06-12 06:22:35.100178361 +0200
@@ -123,35 +123,6 @@ revacccmp(const FTSENT *a, const FTSENT
 }
 
 int
-birthcmp(const FTSENT *a, const FTSENT *b)
-{
-
-	if (b->fts_statp->st_birthtim.tv_sec >
-	    a->fts_statp->st_birthtim.tv_sec)
-		return (1);
-	if (b->fts_statp->st_birthtim.tv_sec <
-	    a->fts_statp->st_birthtim.tv_sec)
-		return (-1);
-	if (b->fts_statp->st_birthtim.tv_nsec >
-	    a->fts_statp->st_birthtim.tv_nsec)
-		return (1);
-	if (b->fts_statp->st_birthtim.tv_nsec <
-	    a->fts_statp->st_birthtim.tv_nsec)
-		return (-1);
-	if (f_samesort)
-		return (strcoll(b->fts_name, a->fts_name));
-	else
-		return (strcoll(a->fts_name, b->fts_name));
-}
-
-int
-revbirthcmp(const FTSENT *a, const FTSENT *b)
-{
-
-	return (birthcmp(b, a));
-}
-
-int
 statcmp(const FTSENT *a, const FTSENT *b)
 {
 
