--- ls/ls.h.orig	2021-04-09 02:24:01.000000000 +0200
+++ ls/ls.h	2021-06-12 06:22:35.112177174 +0200
@@ -42,8 +42,6 @@
 extern long blocksize;		/* block size units */
 
 extern int f_accesstime;	/* use time of last access */
-extern int f_birthtime;	/* use time of file creation */
-extern int f_flags;		/* show flags associated with a file */
 extern int f_humanval;		/* show human-readable file sizes */
 extern int f_label;		/* show MAC label */
 extern int f_inode;		/* print inode */
