--- stat.c.orig	2020-06-29 15:34:05.197755754 -0400
+++ stat.c	2020-07-03 23:54:26.996489189 -0400
@@ -32,6 +32,7 @@
 
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 
 #include <ctype.h>
 #include <err.h>
@@ -45,6 +46,8 @@
 #include <time.h>
 #include <unistd.h>
 
+#include "compat.h"
+
 #define DEF_FORMAT \
 	"%d %i %Sp %l %Su %Sg %r %z \"%Sa\" \"%Sm\" \"%Sc\" " \
 	"%k %b %#Xf %N"
@@ -56,7 +59,7 @@
 	"st_dev=%d st_ino=%i st_mode=%#p st_nlink=%l " \
 	"st_uid=%u st_gid=%g st_rdev=%r st_size=%z " \
 	"st_atime=%a st_mtime=%m st_ctime=%c " \
-	"st_blksize=%k st_blocks=%b st_flags=%f"
+	"st_blksize=%k st_blocks=%b"
 #define LINUX_FORMAT \
 	"  File: \"%N\"%n" \
 	"  Size: %-11z  FileType: %HT%n" \
@@ -119,12 +122,9 @@
 #define SHOW_st_atime	'a'
 #define SHOW_st_mtime	'm'
 #define SHOW_st_ctime	'c'
-#define SHOW_st_btime	'B'
 #define SHOW_st_size	'z'
 #define SHOW_st_blocks	'b'
 #define SHOW_st_blksize	'k'
-#define SHOW_st_flags	'f'
-#define SHOW_st_gen	'v'
 #define SHOW_symlink	'Y'
 #define SHOW_filetype	'T'
 #define SHOW_filename	'N'
@@ -158,9 +158,6 @@
 	int lsF, fmtchar, usestat, fn, nonl, quiet;
 	char *statfmt, *options, *synopsis;
 
-	if (pledge("stdio rpath getpw", NULL) == -1)
-		err(1, "pledge");
-
 	lsF = 0;
 	fmtchar = '\0';
 	usestat = 0;
@@ -444,12 +441,9 @@
 			fmtcase(what, SHOW_st_atime);
 			fmtcase(what, SHOW_st_mtime);
 			fmtcase(what, SHOW_st_ctime);
-			fmtcase(what, SHOW_st_btime);
 			fmtcase(what, SHOW_st_size);
 			fmtcase(what, SHOW_st_blocks);
 			fmtcase(what, SHOW_st_blksize);
-			fmtcase(what, SHOW_st_flags);
-			fmtcase(what, SHOW_st_gen);
 			fmtcase(what, SHOW_symlink);
 			fmtcase(what, SHOW_filetype);
 			fmtcase(what, SHOW_filename);
@@ -611,27 +605,20 @@
 	case SHOW_st_atime:
 		gottime = 1;
 		secs = st->st_atime;
-		nsecs = st->st_atimensec;
+		nsecs = st->st_atim.tv_nsec;
 		/* FALLTHROUGH */
 	case SHOW_st_mtime:
 		if (!gottime) {
 			gottime = 1;
 			secs = st->st_mtime;
-			nsecs = st->st_mtimensec;
+			nsecs = st->st_mtim.tv_nsec;
 		}
 		/* FALLTHROUGH */
 	case SHOW_st_ctime:
 		if (!gottime) {
 			gottime = 1;
 			secs = st->st_ctime;
-			nsecs = st->st_ctimensec;
-		}
-		/* FALLTHROUGH */
-	case SHOW_st_btime:
-		if (!gottime) {
-			gottime = 1;
-			secs = st->__st_birthtimespec.tv_sec;
-			nsecs = st->__st_birthtimespec.tv_nsec;
+			nsecs = st->st_ctim.tv_nsec;
 		}
 		small = (sizeof(secs) == 4);
 		data = secs;
@@ -666,22 +653,6 @@
 		sdata = NULL;
 		formats = FMTF_DECIMAL | FMTF_OCTAL | FMTF_UNSIGNED | FMTF_HEX;
 		if (ofmt == 0)
-			ofmt = FMTF_UNSIGNED;
-		break;
-	case SHOW_st_flags:
-		small = (sizeof(st->st_flags) == 4);
-		data = st->st_flags;
-		sdata = NULL;
-		formats = FMTF_DECIMAL | FMTF_OCTAL | FMTF_UNSIGNED | FMTF_HEX;
-		if (ofmt == 0)
-			ofmt = FMTF_UNSIGNED;
-		break;
-	case SHOW_st_gen:
-		small = (sizeof(st->st_gen) == 4);
-		data = st->st_gen;
-		sdata = NULL;
-		formats = FMTF_DECIMAL | FMTF_OCTAL | FMTF_UNSIGNED | FMTF_HEX;
-		if (ofmt == 0)
 			ofmt = FMTF_UNSIGNED;
 		break;
 	case SHOW_symlink:
