--- id.c.orig	2020-06-25 10:25:58.576582810 -0400
+++ id.c	2020-06-27 13:59:20.364642669 -0400
@@ -30,7 +30,6 @@
  */
 
 #include <sys/types.h>
-#include <sys/socket.h> /* getrtable() lives here */
 
 #include <err.h>
 #include <errno.h>
@@ -41,7 +40,10 @@
 #include <string.h>
 #include <unistd.h>
 #include <limits.h>
-#include <login_cap.h>
+
+#include "compat.h"
+
+extern char *__progname;
 
 void	current(void);
 void	pretty(struct passwd *);
@@ -56,23 +58,20 @@
 {
 	struct group *gr;
 	struct passwd *pw;
-	int ch, cflag, Gflag, gflag, nflag, pflag, Rflag, rflag, uflag;
+	int ch, Gflag, gflag, nflag, pflag, rflag, uflag;
 	uid_t uid;
 	gid_t gid;
 	const char *opts;
 
-	if (pledge("stdio getpw", NULL) == -1)
-		err(1, "pledge");
-
-	cflag = Gflag = gflag = nflag = pflag = Rflag = rflag = uflag = 0;
+	Gflag = gflag = nflag = pflag = rflag = uflag = 0;
 
-	if (strcmp(getprogname(), "groups") == 0) {
+	if (strcmp(__progname, "groups") == 0) {
 		Gflag = 1;
 		nflag = 1;
 		opts = "";
 		if (argc > 2)
 			usage();
-	} else if (strcmp(getprogname(), "whoami") == 0) {
+	} else if (strcmp(__progname, "whoami") == 0) {
 		uflag = 1;
 		nflag = 1;
 		opts = "";
@@ -83,9 +82,6 @@
 
 	while ((ch = getopt(argc, argv, opts)) != -1)
 		switch(ch) {
-		case 'c':
-			cflag = 1;
-			break;
 		case 'G':
 			Gflag = 1;
 			break;
@@ -98,9 +94,6 @@
 		case 'p':
 			pflag = 1;
 			break;
-		case 'R':
-			Rflag = 1;
-			break;
 		case 'r':
 			rflag = 1;
 			break;
@@ -114,7 +107,7 @@
 	argc -= optind;
 	argv += optind;
 
-	switch (cflag + Gflag + gflag + pflag + Rflag + uflag) {
+	switch (Gflag + gflag + pflag + uflag) {
 	case 1:
 		break;
 	case 0:
@@ -128,23 +121,8 @@
 	if (strcmp(opts, "") != 0 && argc > 1)
 		usage();
 
-	if (Rflag) {
-		printf("%d\n", getrtable());
-		exit(0);
-	}
-
 	pw = *argv ? who(*argv) : NULL;
 
-	if (cflag) {
-		if (pw == NULL)
-			pw = getpwuid(getuid());
-		if (pw != NULL && pw->pw_class != NULL && *pw->pw_class != '\0')
-			(void)printf("%s\n", pw->pw_class);
-		else
-			(void)printf("%s\n", LOGIN_DEFCLASS);
-		exit(0);
-	}
-
 	if (gflag) {
 		gid = pw ? pw->pw_gid : rflag ? getgid() : getegid();
 		if (nflag && (gr = getgrgid(gid)))
@@ -218,8 +196,6 @@
 		(void)printf("groups\t");
 		group(NULL, 1);
 	}
-	if (pw != NULL && pw->pw_class != NULL && *pw->pw_class != '\0')
-		(void)printf("class\t%s\n", pw->pw_class);
 }
 
 void
@@ -341,7 +317,7 @@
 	 */
 	if ((pw = getpwnam(u)))
 		return(pw);
-	uid = strtonum(u, 0, UID_MAX, &errstr);
+	uid = strtonum(u, 0, INT_MAX, &errstr);
 	if (!errstr && (pw = getpwuid(uid)))
 		return(pw);
 	errx(1, "%s: No such user", u);
@@ -351,9 +327,9 @@
 void
 usage(void)
 {
-	if (strcmp(getprogname(), "groups") == 0) {
+	if (strcmp(__progname, "groups") == 0) {
 		(void)fprintf(stderr, "usage: groups [user]\n");
-	} else if (strcmp(getprogname(), "whoami") == 0) {
+	} else if (strcmp(__progname, "whoami") == 0) {
 		(void)fprintf(stderr, "usage: whoami\n");
 	} else {
 		(void)fprintf(stderr, "usage: id [user]\n");
@@ -361,7 +337,6 @@
 		(void)fprintf(stderr, "       id -G [-n] [user]\n");
 		(void)fprintf(stderr, "       id -g [-nr] [user]\n");
 		(void)fprintf(stderr, "       id -p [user]\n");
-		(void)fprintf(stderr, "       id -R\n");
 		(void)fprintf(stderr, "       id -u [-nr] [user]\n");
 	}
 	exit(1);
