diff -up comm/comm.c.orig comm/comm.c
--- comm/comm.c.orig	2021-04-27 23:52:35.229551423 -0400
+++ comm/comm.c	2021-04-28 20:48:22.591865579 -0400
@@ -33,7 +33,7 @@
  */
 
 #ifndef lint
-static const char copyright[] =
+__attribute__ ((unused)) static const char copyright[] =
 "@(#) Copyright (c) 1989, 1993, 1994\n\
 	The Regents of the University of California.  All rights reserved.\n";
 #endif
@@ -45,7 +45,6 @@ static char sccsid[] = "From: @(#)comm.c
 #endif
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD$");
 
 #include <err.h>
 #include <limits.h>
