--- bwstring.c.orig	2020-06-27 17:37:25.928824796 -0400
+++ bwstring.c	2020-06-28 13:59:03.684000214 -0400
@@ -462,7 +462,7 @@
  * The strings are nl-ended or zero-ended, depending on the sort setting.
  */
 struct bwstring *
-bwsfgetln(FILE *f, size_t *len, bool zero_ended, struct reader_buffer *rb)
+bwsfgetln(FILE *f, ssize_t *len, bool zero_ended, struct reader_buffer *rb)
 {
 	wint_t eols;
 
@@ -471,9 +471,7 @@
 	if (!zero_ended && (sort_mb_cur_max > 1)) {
 		wchar_t *ret;
 
-		ret = fgetwln(f, len);
-
-		if (ret == NULL) {
+		if (getline((char **) &ret, len, f) == -1) {
 			if (!feof(f))
 				err(2, NULL);
 			return NULL;
@@ -487,9 +485,7 @@
 	} else if (!zero_ended && (sort_mb_cur_max == 1)) {
 		char *ret;
 
-		ret = fgetln(f, len);
-
-		if (ret == NULL) {
+		if (getline(&ret, len, f) == -1) {
 			if (!feof(f))
 				err(2, NULL);
 			return NULL;
