--- bwstring.h.orig	2020-06-27 17:37:25.928824796 -0400
+++ bwstring.h	2020-06-28 13:59:34.300073844 -0400
@@ -91,7 +91,7 @@
 int bwsncmp(const struct bwstring *bws1, const struct bwstring *bws2, size_t offset, size_t len);
 int bwscoll(const struct bwstring *bws1, const struct bwstring *bws2, size_t offset);
 size_t bwsfwrite(struct bwstring *bws, FILE *f, bool zero_ended);
-struct bwstring *bwsfgetln(FILE *file, size_t *len, bool zero_ended, struct reader_buffer *rb);
+struct bwstring *bwsfgetln(FILE *file, ssize_t *len, bool zero_ended, struct reader_buffer *rb);
 
 static inline bwstring_iterator
 bws_begin(struct bwstring *bws)
