--- sort/coll.c.orig	2021-04-09 02:24:12.000000000 +0200
+++ sort/coll.c	2021-06-12 06:22:35.152173219 +0200
@@ -811,7 +811,7 @@ cmpsuffix(unsigned char si1, unsigned ch
  */
 static int
 numcoll_impl(struct key_value *kv1, struct key_value *kv2,
-    size_t offset __unused, bool use_suffix)
+    size_t offset __attribute__((unused)), bool use_suffix)
 {
 	struct bwstring *s1, *s2;
 	wchar_t sfrac1[MAX_NUM_SIZE + 1], sfrac2[MAX_NUM_SIZE + 1];
@@ -995,7 +995,7 @@ randomcoll_init_hint(struct key_value *k
  */
 static int
 randomcoll(struct key_value *kv1, struct key_value *kv2,
-    size_t offset __unused)
+    size_t offset __attribute__((unused)))
 {
 	struct bwstring *s1, *s2;
 	MD5_CTX ctx1, ctx2;
@@ -1043,7 +1043,7 @@ randomcoll(struct key_value *kv1, struct
  */
 static int
 versioncoll(struct key_value *kv1, struct key_value *kv2,
-    size_t offset __unused)
+    size_t offset __attribute__((unused)))
 {
 	struct bwstring *s1, *s2;
 
@@ -1118,7 +1118,7 @@ cmp_nans(double d1, double d2)
  */
 static int
 gnumcoll(struct key_value *kv1, struct key_value *kv2,
-    size_t offset __unused)
+    size_t offset __attribute__((unused)))
 {
 	double d1, d2;
 	int err1, err2;
@@ -1274,7 +1274,7 @@ gnumcoll(struct key_value *kv1, struct k
  * Implements month sort (-M).
  */
 static int
-monthcoll(struct key_value *kv1, struct key_value *kv2, size_t offset __unused)
+monthcoll(struct key_value *kv1, struct key_value *kv2, size_t offset __attribute__((unused)))
 {
 	int val1, val2;
 	bool key1_read, key2_read;
