--- wc.c.orig	2020-06-09 22:34:24.500115693 -0400
+++ wc.c	2020-06-09 22:35:42.331311325 -0400
@@ -42,6 +42,7 @@
 #include <util.h>
 #include <wchar.h>
 #include <wctype.h>
+#include "compat.h"
 
 int64_t	tlinect, twordct, tcharct;
 int	doline, doword, dochar, humanchar, multibyte;
@@ -59,9 +60,6 @@
 
 	setlocale(LC_CTYPE, "");
 
-	if (pledge("stdio rpath", NULL) == -1)
-		err(1, "pledge");
-
 	while ((ch = getopt(argc, argv, "lwchm")) != -1)
 		switch(ch) {
 		case 'l':
