--- touch.c.orig	2020-06-27 17:37:25.933824809 -0400
+++ touch.c	2020-06-30 11:07:47.433704779 -0400
@@ -44,6 +44,8 @@
 #include <time.h>
 #include <unistd.h>
 
+#include "compat.h"
+
 void		stime_arg1(char *, struct timespec *);
 void		stime_arg2(char *, int, struct timespec *);
 void		stime_argd(char *, struct timespec *);
@@ -57,9 +59,6 @@
 	int		 aflag, cflag, mflag, ch, fd, len, rval, timeset;
 	char		*p;
 
-	if (pledge("stdio rpath wpath cpath fattr", NULL) == -1)
-		err(1, "pledge");
-
 	aflag = cflag = mflag = timeset = 0;
 	while ((ch = getopt(argc, argv, "acd:fmr:t:")) != -1)
 		switch (ch) {
