--- arch.c.orig	2020-06-07 21:27:01.582114248 -0400
+++ arch.c	2020-06-07 22:35:41.866603553 -0400
@@ -23,11 +23,12 @@
  * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <sys/param.h>	/* MACHINE MACHINE_ARCH */
-
+#include <err.h>
+#include <libgen.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <sys/utsname.h>
 #include <unistd.h>
 
 static void usage(void);
@@ -37,26 +38,31 @@
 int
 main(int argc, char *argv[])
 {
-	extern char *__progname;
+	char *progname = basename(argv[0]);
 	int short_form = 0, c;
 	char *arch, *opts;
+	struct utsname utsbuf;
+
+	machine = strcmp(progname, "machine") == 0;
+
+	if (uname(&utsbuf) == -1)
+		err(1, "uname(2)");
 
-	machine = strcmp(__progname, "machine") == 0;
 	if (machine) {
-		arch = MACHINE;
+		arch = utsbuf.machine;
 		opts = "a";
 		short_form = 1;
 	} else {
-		arch = MACHINE_ARCH;
+		arch = utsbuf.machine;
 		opts = "ks";
 	}
 	while ((c = getopt(argc, argv, opts)) != -1) {
 		switch (c) {
 		case 'a':
-			arch = MACHINE_ARCH;
+			arch = utsbuf.machine;
 			break;
 		case 'k':
-			arch = MACHINE;
+			arch = utsbuf.machine;
 			break;
 		case 's':
 			short_form = 1;
@@ -68,7 +74,7 @@
 	if (optind != argc)
 		usage();
 
-	printf("%s%s\n", short_form ? "" : "OpenBSD.", arch);
+	printf("%s%s\n", short_form ? "" : utsbuf.sysname, arch);
 	return (0);
 }
 
