--- tsort.c.orig	2020-06-29 15:34:05.201755764 -0400
+++ tsort.c	2020-07-04 09:44:10.984933461 -0400
@@ -305,15 +305,18 @@
 	int 		toggle;
 	struct node 	*a;
 	size_t 		size;
+	ssize_t 	slen;
 	char 		*str;
 
 	toggle = 1;
 	a = NULL;
+	str = NULL;
+	slen = 0;
 
-	while ((str = fgetln(f, &size)) != NULL) {
+	while ((slen = getline(&str, &size, f)) != -1) {
 		char *sentinel;
 
-		sentinel = str + size;
+		sentinel = str + slen;
 		for (;;) {
 			char *e;
 
@@ -358,11 +361,15 @@
 {
 	char 		*str;
 	size_t 		size;
+	ssize_t 	slen;
 
-	while ((str = fgetln(f, &size)) != NULL) {
+	str = NULL;
+	slen = 0;
+
+	while ((slen = getline(&str, &size, f)) != -1) {
 		char *sentinel;
 
-		sentinel = str + size;
+		sentinel = str + slen;
 		for (;;) {
 			char *e;
 			struct node *a;
@@ -992,14 +999,8 @@
 {
 	struct ohash 	pairs;
 
-	if (pledge("stdio rpath", NULL) == -1)
-		err(1, "pledge");
-
 	parse_args(argc, argv, &pairs);
 
-	if (pledge("stdio", NULL) == -1)
-		err(1, "pledge");
-
 	return tsort(&pairs);
 }
 
