--- factor.c.orig	2020-06-07 22:25:25.670165281 -0400
+++ factor.c	2020-06-07 23:15:12.432616889 -0400
@@ -59,6 +59,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
+#include <libgen.h>
 
 #include "primes.h"
 
@@ -78,15 +79,15 @@
 static u_int64_t	usqrt(u_int64_t);
 static void usage(void);
 
+static char *__progname = NULL;
+
 int
 main(int argc, char *argv[])
 {
 	u_int64_t val;
 	int ch;
 	char *p, buf[100];		/* > max number of digits. */
-
-	if (pledge("stdio", NULL) == -1)
-		err(1, "pledge");
+	__progname = basename(argv[0]);
 
 	while ((ch = getopt(argc, argv, "h")) != -1) {
 		switch (ch) {
@@ -305,6 +306,6 @@
 static void
 usage(void)
 {
-	(void)fprintf(stderr, "usage: %s [number ...]\n", getprogname());
+	(void)fprintf(stderr, "usage: %s [number ...]\n", __progname);
 	exit (1);
 }
