--- ln/ln.c.orig	2021-04-09 02:24:01.000000000 +0200
+++ ln/ln.c	2022-07-27 20:25:58.858180989 +0200
@@ -56,6 +56,8 @@ __FBSDID("$FreeBSD$");
 #include <string.h>
 #include <unistd.h>
 
+#include "compat.h"
+
 static int	fflag;			/* Unlink existing files. */
 static int	Fflag;			/* Remove empty directories also. */
 static int	hflag;			/* Check new name for symlink first. */
@@ -148,6 +150,7 @@ main(int argc, char *argv[])
 	switch(argc) {
 	case 0:
 		usage();
+		break;
 		/* NOTREACHED */
 	case 1:				/* ln source */
 		exit(linkit(argv[0], ".", 1));
