--- bc/scan.l.orig	2021-04-09 02:24:12.000000000 +0200
+++ bc/scan.l	2022-07-27 20:24:55.328402751 +0200
@@ -31,6 +31,7 @@ __FBSDID("$FreeBSD$");
 #include "extern.h"
 #include "bc.h"
 #include "pathnames.h"
+#include "compat.h"
 
 int		lineno;
 bool		interactive;
@@ -242,7 +243,7 @@ add_str(const char *str)
 
 /* ARGSUSED */
 void
-abort_line(int sig __unused)
+abort_line(int sig __attribute__((unused)))
 {
 	static const char str1[] = "[\n]P\n";
 	static const char str2[] = "[^C\n]P\n";
@@ -265,7 +266,7 @@ abort_line(int sig __unused)
  * line after a ^C.
  */
 unsigned char
-bc_eof(EditLine *e, int ch __unused)
+bc_eof(EditLine *e, int ch __attribute__((unused)))
 {
 	const struct lineinfo *info = el_line(e);
 
