--- bc/scan.l.orig	2021-06-12 06:26:11.490778976 +0200
+++ bc/scan.l	2021-06-12 06:47:02.271086318 +0200
@@ -191,7 +191,8 @@ ALPHANUM	[a-z_0-9]
 			char *p = malloc(yyleng + 2);
 			if (p == NULL)
 				err(1, NULL);
-			strlcpy(p, yytext, yyleng + 1);
+			strncpy(p, yytext, yyleng + 1);
+			yytext[yyleng] = '\0';
 			yylval.astr = p;
 			return LETTER;
 		}
@@ -237,12 +238,13 @@ add_str(const char *str)
 		strbuf_sz = newsize;
 		strbuf = p;
 	}
-	strlcat(strbuf, str, strbuf_sz);
+	strncat(strbuf, str, strbuf_sz);
+	strbuf[strbuf_sz - 1] = '\0';
 }
 
 /* ARGSUSED */
 void
-abort_line(int sig __unused)
+abort_line(int sig __attribute__((unused)))
 {
 	static const char str1[] = "[\n]P\n";
 	static const char str2[] = "[^C\n]P\n";
@@ -265,7 +267,7 @@ abort_line(int sig __unused)
  * line after a ^C.
  */
 unsigned char
-bc_eof(EditLine *e, int ch __unused)
+bc_eof(EditLine *e, int ch __attribute__((unused)))
 {
 	const struct lineinfo *info = el_line(e);
 
