diff -up basename/basename.c.orig basename/basename.c
--- basename/basename.c.orig	2021-04-30 09:00:42.861377850 -0400
+++ basename/basename.c	2021-04-30 09:02:26.314612984 -0400
@@ -44,7 +44,6 @@ static char sccsid[] = "@(#)basename.c	8
 #include <sys/cdefs.h>
 __FBSDID("$FreeBSD$");
 
-#include <capsicum_helpers.h>
 #include <err.h>
 #include <libgen.h>
 #include <limits.h>
@@ -67,9 +66,6 @@ main(int argc, char **argv)
 
 	setlocale(LC_ALL, "");
 
-	if (caph_limit_stdio() < 0 || caph_enter() < 0)
-		err(1, "capsicum");
-
 	aflag = 0;
 	suffix = NULL;
 	suffixlen = 0;
