--- printenv/printenv.c.orig	2021-04-09 02:24:12.000000000 +0200
+++ printenv/printenv.c	2021-06-12 06:22:35.136174801 +0200
@@ -46,7 +46,6 @@ __FBSDID("$FreeBSD$");
 
 #include <sys/types.h>
 
-#include <capsicum_helpers.h>
 #include <err.h>
 #include <stdlib.h>
 #include <stdio.h>
@@ -69,9 +68,6 @@ main(int argc, char *argv[])
 	size_t len;
 	int ch;
 
-	if (caph_limit_stdio() < 0 || caph_enter() < 0)
-		err(1, "capsicum");
-
 	while ((ch = getopt(argc, argv, "")) != -1)
 		switch(ch) {
 		case '?':
