--- dc/dc.c.orig	2021-04-09 02:24:12.000000000 +0200
+++ dc/dc.c	2021-06-12 06:22:35.016186667 +0200
@@ -22,7 +22,6 @@ __FBSDID("$FreeBSD$");
 
 #include <sys/stat.h>
 
-#include <capsicum_helpers.h>
 #include <ctype.h>
 #include <err.h>
 #include <errno.h>
@@ -135,19 +134,12 @@ main(int argc, char *argv[])
 		if (fd < 0)
 			err(1, "cannot open file %s", argv[0]);
 
-		if (caph_limit_stream(fd, CAPH_READ) < 0 ||
-		    caph_limit_stdio() < 0 ||
-		    caph_enter() < 0)
-			err(1, "capsicum");
-
 		procfd(fd, argv[0]);
 		preproc_done = true;
 	}
 	if (preproc_done)
 		return (0);
 
-	if (caph_limit_stdio() < 0 || caph_enter())
-		err(1, "capsicum");
 	src_setstream(&src, stdin);
 	reset_bmachine(&src);
 	eval();
