diff -up chown/chown.c.orig chown/chown.c
--- chown/chown.c.orig	2021-04-27 23:52:35.229551423 -0400
+++ chown/chown.c	2021-04-30 18:27:11.625136464 -0400
@@ -73,14 +73,6 @@ static uid_t uid;
 static gid_t gid;
 static int ischown;
 static const char *gname;
-static volatile sig_atomic_t siginfo;
-
-static void
-siginfo_handler(int sig __unused)
-{
-
-	siginfo = 1;
-}
 
 int
 main(int argc, char **argv)
@@ -132,8 +124,6 @@ main(int argc, char **argv)
 	if (argc < 2)
 		usage();
 
-	(void)signal(SIGINFO, siginfo_handler);
-
 	if (Rflag) {
 		if (hflag && (Hflag || Lflag))
 			errx(1, "the -R%c and -h options may not be "
@@ -204,10 +194,6 @@ main(int argc, char **argv)
 		default:
 			break;
 		}
-		if (siginfo) {
-			print_info(p, 2);
-			siginfo = 0;
-		}
 		if ((uid == (uid_t)-1 || uid == p->fts_statp->st_uid) &&
 		    (gid == (gid_t)-1 || gid == p->fts_statp->st_gid))
 			continue;
