--- chown/chown.c.orig	2021-04-09 02:24:13.000000000 +0200
+++ chown/chown.c	2021-10-27 16:35:31.524720457 +0200
@@ -62,6 +62,8 @@ __FBSDID("$FreeBSD$");
 #include <string.h>
 #include <unistd.h>
 
+#include "compat.h"
+
 static void	a_gid(const char *);
 static void	a_uid(const char *);
 static void	chownerr(const char *);
@@ -76,7 +78,7 @@ static const char *gname;
 static volatile sig_atomic_t siginfo;
 
 static void
-siginfo_handler(int sig __unused)
+siginfo_handler(int sig __attribute__((unused)))
 {
 
 	siginfo = 1;
