diff -up dirname/dirname.c.orig dirname/dirname.c
--- dirname/dirname.c.orig	2021-04-27 23:52:35.231551429 -0400
+++ dirname/dirname.c	2021-04-29 08:37:09.697144814 -0400
@@ -30,18 +30,16 @@
  */
 
 #ifndef lint
-static const char copyright[] =
+__attribute__ ((unused)) static const char copyright[] =
 "@(#) Copyright (c) 1991, 1993, 1994\n\
 	The Regents of the University of California.  All rights reserved.\n";
 #endif /* not lint */
 
 #ifndef lint
-static const char sccsid[] = "@(#)dirname.c	8.4 (Berkeley) 5/4/95";
+__attribute__ ((unused)) static const char sccsid[] = "@(#)dirname.c	8.4 (Berkeley) 5/4/95";
 #endif /* not lint */
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD$");
 
-#include <capsicum_helpers.h>
 #include <err.h>
 #include <libgen.h>
 #include <stdio.h>
@@ -56,9 +54,6 @@ main(int argc, char **argv)
 	char *p;
 	int ch;
 
-	if (caph_limit_stdio() < 0 || caph_enter() < 0)
-		err(1, "capsicum");
-
 	while ((ch = getopt(argc, argv, "")) != -1)
 		switch(ch) {
 		case '?':
