diff -up dirname/dirname.c.orig dirname/dirname.c
--- dirname/dirname.c.orig	2021-04-30 09:15:28.099389935 -0400
+++ dirname/dirname.c	2021-04-30 09:19:09.741893716 -0400
@@ -41,7 +41,6 @@ static const char sccsid[] = "@(#)dirnam
 #include <sys/cdefs.h>
 __FBSDID("$FreeBSD$");
 
-#include <capsicum_helpers.h>
 #include <err.h>
 #include <libgen.h>
 #include <stdio.h>
@@ -56,9 +55,6 @@ main(int argc, char **argv)
 	char *p;
 	int ch;
 
-	if (caph_limit_stdio() < 0 || caph_enter() < 0)
-		err(1, "capsicum");
-
 	while ((ch = getopt(argc, argv, "")) != -1)
 		switch(ch) {
 		case '?':
