--- seq/seq.c.orig	2021-04-09 02:24:12.000000000 +0200
+++ seq/seq.c	2021-06-12 06:22:35.148173614 +0200
@@ -52,6 +52,8 @@ __FBSDID("$FreeBSD$");
 #define ISEXP(c)	((int)(c) == 'e' || (int)(c) == 'E')
 #define ISODIGIT(c)	((int)(c) >= '0' && (int)(c) <= '7')
 
+extern char *__progname;
+
 /* Globals */
 
 static const char *decimal_point = ".";	/* default */
@@ -143,7 +145,7 @@ main(int argc, char *argv[])
 	if (errflg) {
 		fprintf(stderr,
 		    "usage: %s [-w] [-f format] [-s string] [-t string] [first [incr]] last\n",
-		    getprogname());
+		    __progname);
 		exit(1);
 	}
 
