--- env/envopts.c.orig	2021-04-09 02:24:12.000000000 +0200
+++ env/envopts.c	2021-06-12 06:22:35.052183107 +0200
@@ -33,6 +33,7 @@
 #include <sys/cdefs.h>
 __FBSDID("$FreeBSD$");
 
+#include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/param.h>
 #include <err.h>
@@ -418,7 +419,8 @@ expand_vars(int in_thisarg, char **thisa
 	*src_p = vend;
 	namelen = vend - vbegin + 1;
 	vname = malloc(namelen);
-	strlcpy(vname, vbegin, namelen);
+	strncpy(vname, vbegin, namelen);
+	vname[namelen - 1] = '\0';
 	vvalue = getenv(vname);
 	if (vvalue == NULL || *vvalue == '\0') {
 		if (env_verbosity > 2)
