--- hexdump/hexdump.c.orig	2021-07-02 01:43:34.555292536 +0200
+++ hexdump/hexdump.c	2021-07-02 01:43:57.033036817 +0200
@@ -44,8 +44,6 @@ static char sccsid[] = "@(#)hexdump.c	8.
 __FBSDID("$FreeBSD$");
 
 #include <sys/types.h>
-#include <sys/capsicum.h>
-#include <capsicum_helpers.h>
 #include <err.h>
 #include <locale.h>
 #include <stdlib.h>
@@ -81,14 +79,6 @@ main(int argc, char *argv[])
 	for (tfs = fshead; tfs; tfs = tfs->nextfs)
 		rewrite(tfs);
 
-	/*
-	 * Cache NLS data, for strerror, for err(3), before entering capability
-	 * mode.
-	 */
-	caph_cache_catpages();
-	if (caph_limit_stdio() < 0)
-		err(1, "capsicum");
-
 	(void)next(argv);
 	display();
 	exit(exitval);
