--- tty/tty.c.orig	2021-04-27 23:52:35.241551457 -0400
+++ tty/tty.c	2021-05-06 09:30:17.982010122 -0400
@@ -48,6 +48,8 @@ __FBSDID("$FreeBSD$");
 #include <stdlib.h>
 #include <unistd.h>
 
+extern char *__progname;
+
 static void usage(void);
 
 int
@@ -77,6 +79,6 @@ main(int argc, char *argv[])
 static void
 usage(void)
 {
-	fprintf(stderr, "usage: %s [-s]\n", getprogname());
+	fprintf(stderr, "usage: %s [-s]\n", __progname);
 	exit(2);
 }
