diff -up fmt/fmt.c.orig fmt/fmt.c
--- fmt/fmt.c.orig	2021-04-27 23:52:35.233551434 -0400
+++ fmt/fmt.c	2021-04-30 08:50:23.073969157 -0400
@@ -185,6 +185,7 @@ __FBSDID("$FreeBSD$");
 #include <unistd.h>
 #include <wchar.h>
 #include <wctype.h>
+#include <stdint.h>
 
 /* Something that, we hope, will never be a genuine line length,
  * indentation etc.
@@ -380,7 +381,7 @@ main(int argc, char *argv[])
 		goal_length = 65;
 	if (max_length == 0)
 		max_length = goal_length + 10;
-	if (max_length >= SIZE_T_MAX / sizeof(wchar_t))
+	if (max_length >= SIZE_MAX / sizeof(wchar_t))
 		errx(EX_USAGE, "max length too large");
 	/* really needn't be longer */
 	output_buffer = XMALLOC((max_length + 1) * sizeof(wchar_t));
