--- mkdir.c.orig	2020-06-26 10:20:13.218148877 -0400
+++ mkdir.c	2020-06-27 16:11:11.376964685 -0400
@@ -40,6 +40,8 @@
 #include <string.h>
 #include <unistd.h>
 
+#include "compat.h"
+
 extern char *__progname;
 
 int	mkpath(char *, mode_t, mode_t);
@@ -79,11 +81,6 @@
 	argc -= optind;
 	argv += optind;
 
-	if ((mode & (S_ISUID | S_ISGID | S_ISTXT)) == 0) {
-		if (pledge("stdio rpath cpath fattr", NULL) == -1)
-			err(1, "pledge");
-	}
-
 	if (*argv == NULL)
 		usage();
 
