--- stty/gfmt.c.orig	2021-06-30 17:30:57.488135019 +0200
+++ stty/gfmt.c	2021-06-30 17:00:32.011330012 +0200
@@ -40,7 +40,9 @@ __FBSDID("$FreeBSD$");
 #include <err.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <unistd.h>
 #include <string.h>
+#include <termios.h>
 
 #include "stty.h"
 #include "extern.h"
@@ -57,7 +59,7 @@ gerr(const char *s)
 }
 
 void
-gprint(struct termios *tp, struct winsize *wp __unused, int ldisc __unused)
+gprint(struct termios *tp, struct winsize *wp __attribute__((unused)), int ldisc __attribute__((unused)))
 {
 	struct cchar *cp;
 
@@ -99,7 +101,7 @@ gread(struct termios *tp, char *s)
 		}
 		if (CHK("ispeed")) {
 			tmp = strtoul(ep, NULL, 10);
-			tp->c_ispeed = tmp;
+			cfsetispeed(tp, tmp);
 			continue;
 		}
 		if (CHK("lflag")) {
@@ -112,7 +114,7 @@ gread(struct termios *tp, char *s)
 		}
 		if (CHK("ospeed")) {
 			tmp = strtoul(ep, NULL, 10);
-			tp->c_ospeed = tmp;
+			cfsetospeed(tp, tmp);
 			continue;
 		}
 		for (cp = cchars1; cp->name != NULL; ++cp)
