--- modes.c.orig	2020-06-30 11:07:47.431704771 -0400
+++ modes.c	2020-07-03 15:44:34.391398655 -0400
@@ -78,8 +78,6 @@
 	{ "-clocal",	0, CLOCAL },
 	{ "crtscts",	CRTSCTS, 0 },
 	{ "-crtscts",	0, CRTSCTS },
-	{ "mdmbuf",	MDMBUF, 0 },
-	{ "-mdmbuf",	0, MDMBUF },
 	{ NULL },
 };
 
@@ -136,8 +134,8 @@
 	{ "-echoke",	0, ECHOKE },
 	{ "crtkill",	ECHOKE, 0 },
 	{ "-crtkill",	0, ECHOKE },
-	{ "altwerase",	ALTWERASE, 0 },
-	{ "-altwerase",	0, ALTWERASE },
+	{ "altwerase",	VWERASE, 0 },
+	{ "-altwerase",	0, VWERASE },
 	{ "iexten",	IEXTEN, 0 },
 	{ "-iexten",	0, IEXTEN },
 	{ "echonl",	ECHONL, 0 },
@@ -166,10 +164,6 @@
 	{ "-crt",	ECHOK, ECHOE|ECHOKE|ECHOCTL },
 	{ "newcrt",	ECHOE|ECHOKE|ECHOCTL, ECHOK|ECHOPRT },
 	{ "-newcrt",	ECHOK, ECHOE|ECHOKE|ECHOCTL },
-	{ "nokerninfo",	NOKERNINFO, 0 },
-	{ "-nokerninfo",0, NOKERNINFO },
-	{ "kerninfo",	0, NOKERNINFO },
-	{ "-kerninfo",	NOKERNINFO, 0 },
 	{ "xcase",	XCASE, 0 },
 	{ "-xcase",	0, XCASE },
 	{ NULL },
@@ -190,12 +184,10 @@
 	{ "-onlret",	0, ONLRET },
 	{ "onocr",	ONOCR, 0 },
 	{ "-onocr",	0, ONOCR },
-	{ "tabs",	0, OXTABS },		/* "preserve" tabs */
-	{ "-tabs",	OXTABS, 0 },
-	{ "oxtabs",	OXTABS, 0 },
-	{ "-oxtabs",	0, OXTABS },
-	{ "onoeot",	ONOEOT, 0 },
-	{ "-onoeot",	0, ONOEOT },
+	{ "tabs",	0, XTABS },		/* "preserve" tabs */
+	{ "-tabs",	XTABS, 0 },
+	{ "oxtabs",	XTABS, 0 },
+	{ "-oxtabs",	0, XTABS },
 	{ NULL },
 };
 
