--- stty.c.orig	2020-06-30 11:07:47.432704775 -0400
+++ stty.c	2020-07-03 15:51:17.478619688 -0400
@@ -47,6 +47,8 @@
 #include "stty.h"
 #include "extern.h"
 
+#include "compat.h"
+
 int
 main(int argc, char *argv[])
 {
@@ -99,23 +101,15 @@
 	case POSIX:
 		if (*argv)
 			errx(1, "either display or modify");
-		if (pledge("stdio", NULL) == -1)
-			err(1, "pledge");
 		print(&i.t, &i.win, i.ldisc, fmt);
 		break;
 	case GFLAG:
 		if (*argv)
 			errx(1, "either display or modify");
-		if (pledge("stdio", NULL) == -1)
-			err(1, "pledge");
 		gprint(&i.t, &i.win, i.ldisc);
 		break;
 	}
 
-	/*
-	 * Cannot pledge, because of "extproc", "ostart" and "ostop"
-	 */
-
 	for (i.set = i.wset = 0; *argv; ++argv) {
 		if (ksearch(&argv, &i))
 			continue;
