diff -up echo/echo.c.orig echo/echo.c
--- echo/echo.c.orig	2021-04-27 23:52:35.231551429 -0400
+++ echo/echo.c	2021-04-29 08:39:27.186602980 -0400
@@ -41,13 +41,11 @@ static char sccsid[] = "@(#)echo.c	8.1 (
 #endif /* not lint */
 #endif
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD$");
 
 #include <sys/types.h>
 #include <sys/uio.h>
 
 #include <assert.h>
-#include <capsicum_helpers.h>
 #include <err.h>
 #include <errno.h>
 #include <limits.h>
@@ -59,7 +57,7 @@ __FBSDID("$FreeBSD$");
  * Report an error and exit.
  * Use it instead of err(3) to avoid linking-in stdio.
  */
-static __dead2 void
+static void
 errexit(const char *prog, const char *reason)
 {
 	char *errstr = strerror(errno);
@@ -82,9 +80,6 @@ main(int argc, char *argv[])
 	char newline[] = "\n";
 	char *progname = argv[0];
 
-	if (caph_limit_stdio() < 0 || caph_enter() < 0)
-		err(1, "capsicum");
-
 	/* This utility may NOT do getopt(3) option parsing. */
 	if (*++argv && !strcmp(*argv, "-n")) {
 		++argv;
