--- nice.c.orig	2020-06-27 17:37:25.924824786 -0400
+++ nice.c	2020-06-27 17:37:53.799895590 -0400
@@ -39,6 +39,8 @@
 #include <stdlib.h>
 #include <unistd.h>
 
+#include "compat.h"
+
 #define	DEFNICE	10
 
 static void usage(void);
@@ -50,9 +52,6 @@
 	int prio = DEFNICE;
 	int c;
 
-	if (pledge("stdio exec proc", NULL) == -1)
-		err(1, "pledge");
-
 	/* handle obsolete -number syntax */
 	if (argc > 1 && argv[1][0] == '-' &&
 	    isdigit((unsigned char)argv[1][1])) {
@@ -87,9 +86,6 @@
 	if (setpriority(PRIO_PROCESS, 0, prio))
 		warn("setpriority");
 
-	if (pledge("stdio exec", NULL) == -1)
-		err(1, "pledge");
-
 	execvp(argv[0], &argv[0]);
 	err((errno == ENOENT) ? 127 : 126, "%s", argv[0]);
 }
