--- du.c.orig	2020-06-25 09:46:14.128014551 -0400
+++ du.c	2020-06-25 10:18:54.961211994 -0400
@@ -44,10 +44,12 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <sys/tree.h>
 #include <unistd.h>
 #include <util.h>
 
+#include <sys/param.h>
+#include "compat.h"
+#include "tree.h"
 
 int	 linkchk(FTSENT *);
 void	 prtout(int64_t, char *, int);
@@ -66,9 +68,6 @@
 	char **save;
 	const char *errstr;
 
-	if (pledge("stdio rpath", NULL) == -1)
-		err(1, "pledge");
-
 	save = argv;
 	Hflag = Lflag = cflag = hflag = kflag = listfiles = 0;
 	totalblocks = 0;
@@ -186,7 +185,8 @@
 		case FTS_DNR:			/* Warn, continue. */
 		case FTS_ERR:
 		case FTS_NS:
-			warnc(p->fts_errno, "%s", p->fts_path);
+			errno = p->fts_errno;
+			warn("%s", p->fts_path);
 			rval = 1;
 			break;
 		default:
