diff -up true/true.c.orig true/true.c
--- true/true.c.orig	2021-04-08 20:24:13.000000000 -0400
+++ true/true.c	2021-04-27 23:39:49.923430482 -0400
@@ -32,13 +32,13 @@
  */
 
 #ifndef lint
-static const char copyright[] =
+__attribute__ ((unused)) static const char copyright[] =
 "@(#) Copyright (c) 1988, 1993\n\
 	The Regents of the University of California.  All rights reserved.\n";
 #endif /* not lint */
 
 #ifndef lint
-static const char sccsid[] = "@(#)true.c	8.1 (Berkeley) 6/9/93";
+__attribute__ ((unused)) static const char sccsid[] = "@(#)true.c	8.1 (Berkeley) 6/9/93";
 #endif /* not lint */
 
 int
