--- cut.c.orig	2020-06-25 09:46:14.138014574 -0400
+++ cut.c	2020-06-25 10:03:05.805428719 -0400
@@ -44,6 +44,8 @@
 #include <string.h>
 #include <unistd.h>
 
+#include "compat.h"
+
 char	dchar[5];
 int	dlen;
 
@@ -69,9 +71,6 @@
 
 	setlocale(LC_CTYPE, "");
 
-	if (pledge("stdio rpath", NULL) == -1)
-		err(1, "pledge");
-
 	dchar[0] = '\t';		/* default delimiter */
 	dchar[1] = '\0';
 	dlen = 1;
@@ -142,9 +141,6 @@
 			}
 		}
 	else {
-		if (pledge("stdio", NULL) == -1)
-			err(1, "pledge");
-
 		fcn(stdin, "stdin");
 	}
 	exit(rval);
