--- fold.c.orig	2020-06-25 10:25:58.575582807 -0400
+++ fold.c	2020-06-25 19:29:23.185524533 -0400
@@ -43,6 +43,8 @@
 #include <unistd.h>
 #include <wchar.h>
 
+#include "compat.h"
+
 #define	DEFLINEWIDTH	80
 
 static void fold(unsigned int);
@@ -61,9 +63,6 @@
 
 	setlocale(LC_CTYPE, "");
 
-	if (pledge("stdio rpath", NULL) == -1)
-		err(1, "pledge");
-
 	width = 0;
 	lastch = '\0';
 	prevoptind = 1;
@@ -108,8 +107,6 @@
 		width = DEFLINEWIDTH;
 
 	if (!*argv) {
-		if (pledge("stdio", NULL) == -1)
-			err(1, "pledge");
 		fold(width);
 	} else {
 		for (; *argv; ++argv) {
