--- mv.c.orig	2020-06-26 10:20:13.220148881 -0400
+++ mv.c	2020-06-27 17:33:20.109180493 -0400
@@ -37,6 +37,7 @@
 #include <sys/wait.h>
 #include <sys/stat.h>
 #include <sys/mount.h>
+#include <sys/statvfs.h>
 
 #include <err.h>
 #include <errno.h>
@@ -49,6 +50,8 @@
 #include <pwd.h>
 #include <grp.h>
 
+#include "compat.h"
+
 extern char *__progname;
 
 int fflg, iflg, vflg;
@@ -224,14 +227,14 @@
 
 	/* Disallow moving a mount point. */
 	if (S_ISDIR(fsb.st_mode)) {
-		struct statfs sfs;
+		struct statvfs sfs;
 		char path[PATH_MAX];
 
 		if (realpath(from, path) == NULL) {
 			warnx("cannot resolve %s", from);
 			return (1);
 		}
-		if (!statfs(path, &sfs) && !strcmp(path, sfs.f_mntonname)) {
+		if (!statvfs(path, &sfs)) {
 			warnx("cannot rename a mount point");
 			return (1);
 		}
@@ -309,28 +312,18 @@
 
 	if (badchown) {
 		if ((sbp->st_mode & (S_ISUID|S_ISGID)))  {
-			warnc(serrno,
-			    "%s: set owner/group; not setting setuid/setgid",
+			errno = serrno;
+			warn("%s: set owner/group; not setting setuid/setgid",
 			    to);
 			sbp->st_mode &= ~(S_ISUID|S_ISGID);
-		} else if (!fflg)
-			warnc(serrno, "%s: set owner/group", to);
+		} else if (!fflg) {
+			errno = serrno;
+			warn("%s: set owner/group", to);
+		}
 	}
 	if (fchmod(to_fd, sbp->st_mode))
 		warn("%s: set mode", to);
 
-	/*
-	 * XXX
-	 * NFS doesn't support chflags; ignore errors unless there's reason
-	 * to believe we're losing bits.  (Note, this still won't be right
-	 * if the server supports flags and we were trying to *remove* flags
-	 * on a file that we copied, i.e., that we didn't create.)
-	 */
-	errno = 0;
-	if (fchflags(to_fd, sbp->st_flags))
-		if (errno != EOPNOTSUPP || sbp->st_flags != 0)
-			warn("%s: set flags", to);
-
 	ts[0] = sbp->st_atim;
 	ts[1] = sbp->st_mtim;
 	if (futimens(to_fd, ts))
