--- expr/expr.y.orig	2021-04-09 02:24:01.000000000 +0200
+++ expr/expr.y	2021-06-12 06:47:02.271086318 +0200
@@ -270,8 +270,7 @@ main(int argc, char *argv[])
 	int c;
 
 	setlocale(LC_ALL, "");
-	if (getenv("EXPR_COMPAT") != NULL
-	    || check_utility_compat("expr")) {
+	if (getenv("EXPR_COMPAT") != NULL) {
 		av = argv + 1;
 		nonposix = 1;
 	} else {
@@ -299,7 +298,7 @@ main(int argc, char *argv[])
 }
 
 int
-yyerror(const char *s __unused)
+yyerror(const char *s __attribute__((unused)))
 {
 	errx(ERR_EXIT, "syntax error");
 }
