--- yes/yes.c.orig	2021-04-09 02:24:13.000000000 +0200
+++ yes/yes.c	2021-06-12 06:22:35.208167681 +0200
@@ -43,7 +43,6 @@ static const char rcsid[] = "$FreeBSD$";
 #endif
 #endif /* not lint */
 
-#include <capsicum_helpers.h>
 #include <err.h>
 #include <stdio.h>
 #include <string.h>
@@ -60,9 +59,6 @@ main(int argc, char **argv)
 	size_t more;
 	ssize_t ret;
 
-	if (caph_limit_stdio() < 0 || caph_enter() < 0)
-		err(1, "capsicum");
-
 	if (argc > 1) {
 		exp = argv[1];
 		explen = strlen(exp) + 1;
