--- uniq.c.orig	2020-06-24 22:00:39.344211781 -0400
+++ uniq.c	2020-06-24 22:17:33.684350714 -0400
@@ -45,6 +45,8 @@
 #include <wchar.h>
 #include <wctype.h>
 
+#include "compat.h"
+
 #define	MAXLINELEN	(8 * 1024)
 
 int cflag, dflag, iflag, uflag;
@@ -66,9 +68,6 @@
 
 	setlocale(LC_CTYPE, "");
 
-	if (pledge("stdio rpath wpath cpath", NULL) == -1)
-		err(1, "pledge");
-
 	obsolete(argv);
 	while ((ch = getopt(argc, argv, "cdf:is:u")) != -1) {
 		const char *errstr;
@@ -130,9 +129,6 @@
 		usage();
 	}
 
-	if (pledge("stdio", NULL) == -1)
-		err(1, "pledge");
-
 	prevline = malloc(MAXLINELEN);
 	thisline = malloc(MAXLINELEN);
 	if (prevline == NULL || thisline == NULL)
