--- uniq/uniq.c.orig	2021-04-09 02:24:13.000000000 +0200
+++ uniq/uniq.c	2021-06-12 06:22:35.192169263 +0200
@@ -46,9 +46,6 @@ static const char rcsid[] =
   "$FreeBSD$";
 #endif /* not lint */
 
-#include <sys/capsicum.h>
-
-#include <capsicum_helpers.h>
 #include <ctype.h>
 #include <err.h>
 #include <errno.h>
@@ -103,7 +100,6 @@ main (int argc, char *argv[])
 	size_t prevbuflen, thisbuflen, b1;
 	char *prevline, *thisline, *p;
 	const char *ifn;
-	cap_rights_t rights;
 
 	(void) setlocale(LC_ALL, "");
 
@@ -159,32 +155,8 @@ main (int argc, char *argv[])
 	ofp = stdout;
 	if (argc > 0 && strcmp(argv[0], "-") != 0)
 		ifp = file(ifn = argv[0], "r");
-	cap_rights_init(&rights, CAP_FSTAT, CAP_READ);
-	if (caph_rights_limit(fileno(ifp), &rights) < 0)
-		err(1, "unable to limit rights for %s", ifn);
-	cap_rights_init(&rights, CAP_FSTAT, CAP_WRITE);
 	if (argc > 1)
 		ofp = file(argv[1], "w");
-	else
-		cap_rights_set(&rights, CAP_IOCTL);
-	if (caph_rights_limit(fileno(ofp), &rights) < 0) {
-		err(1, "unable to limit rights for %s",
-		    argc > 1 ? argv[1] : "stdout");
-	}
-	if (cap_rights_is_set(&rights, CAP_IOCTL)) {
-		unsigned long cmd;
-
-		cmd = TIOCGETA; /* required by isatty(3) in printf(3) */
-
-		if (caph_ioctls_limit(fileno(ofp), &cmd, 1) < 0) {
-			err(1, "unable to limit ioctls for %s",
-			    argc > 1 ? argv[1] : "stdout");
-		}
-	}
-
-	caph_cache_catpages();
-	if (caph_enter() < 0)
-		err(1, "unable to enter capability mode");
 
 	prevbuflen = thisbuflen = 0;
 	prevline = thisline = NULL;
