--- dd/args.c.orig	2021-04-09 02:24:01.000000000 +0200
+++ dd/args.c	2021-06-12 06:22:35.024185876 +0200
@@ -55,6 +55,8 @@ __FBSDID("$FreeBSD$");
 #include "dd.h"
 #include "extern.h"
 
+#include "compat.h"
+
 static int	c_arg(const void *, const void *);
 static int	c_conv(const void *, const void *);
 static int	c_iflag(const void *, const void *);
@@ -214,8 +216,10 @@ f_count(char *arg)
 	uintmax_t res;
 
 	res = get_num(arg);
-	if (res == UINTMAX_MAX)
-		errc(1, ERANGE, "%s", oper);
+	if (res == UINTMAX_MAX) {
+		errno = ERANGE;
+		err(1, "%s", oper);
+	}
 	if (res == 0)
 		cpy_cnt = UINTMAX_MAX;
 	else
