--- dd/misc.c.orig	2021-04-09 02:24:01.000000000 +0200
+++ dd/misc.c	2021-10-27 16:18:58.887798082 +0200
@@ -46,7 +46,6 @@ __FBSDID("$FreeBSD$");
 #include <err.h>
 #include <errno.h>
 #include <inttypes.h>
-#include <libutil.h>
 #include <signal.h>
 #include <stdio.h>
 #include <stdlib.h>
@@ -57,6 +56,8 @@ __FBSDID("$FreeBSD$");
 #include "dd.h"
 #include "extern.h"
 
+#include "compat.h"
+
 double
 secs_elapsed(void)
 {
@@ -133,7 +134,7 @@ progress(void)
 
 /* ARGSUSED */
 void
-siginfo_handler(int signo __unused)
+siginfo_handler(int signo __attribute__((unused)))
 {
 
 	need_summary = 1;
@@ -141,7 +142,7 @@ siginfo_handler(int signo __unused)
 
 /* ARGSUSED */
 void
-sigalarm_handler(int signo __unused)
+sigalarm_handler(int signo __attribute__((unused)))
 {
 
 	need_progress = 1;
