--- misc.c.orig	2020-06-07 22:25:25.957165951 -0400
+++ misc.c	2020-06-07 23:03:04.407051018 -0400
@@ -45,6 +45,20 @@
 #include "dd.h"
 #include "extern.h"
 
+/*
+ * From <sys/time.h> on OpenBSD.  Not used in any other bsdutils commands
+ * so just putting it in this file.
+ */
+#define timespecsub(tsp, usp, vsp)                                \
+        do {                                                      \
+                (vsp)->tv_sec = (tsp)->tv_sec - (usp)->tv_sec;    \
+                (vsp)->tv_nsec = (tsp)->tv_nsec - (usp)->tv_nsec; \
+                if ((vsp)->tv_nsec < 0) {                         \
+                        (vsp)->tv_sec--;                          \
+                        (vsp)->tv_nsec += 1000000000L;            \
+                }                                                 \
+        } while (0)
+
 void
 summary(void)
 {
@@ -82,15 +96,6 @@
 }
 
 void
-summaryx(int notused)
-{
-	int save_errno = errno;
-
-	summary();
-	errno = save_errno;
-}
-
-void
 terminate(int signo)
 {
 	summary();
