--- who.c.orig	2020-06-09 22:20:03.821440999 -0400
+++ who.c	2020-06-09 22:27:35.233814625 -0400
@@ -45,6 +45,7 @@
 #include <time.h>
 #include <err.h>
 #include <locale.h>
+#include "compat.h"
 
 void  output(struct utmp *);
 void  output_labels(void);
@@ -74,9 +75,6 @@
 
 	setlocale(LC_ALL, "");
 
-	if (pledge("stdio unveil rpath getpw", NULL) == -1)
-		err(1, "pledge");
-
 	if ((mytty = ttyname(0))) {
 		/* strip any directory component */
 		if ((t = strrchr(mytty, '/')))
@@ -122,12 +120,8 @@
 	if (show_labels)
 		output_labels();
 
-	if (unveil(_PATH_UTMP, "r") == -1)
-		err(1, "unveil");
 	switch (argc) {
 	case 0:					/* who */
-		if (pledge("stdio rpath getpw", NULL) == -1)
-			err(1, "pledge");
 		ufp = file(_PATH_UTMP);
 
 		if (only_current_term) {
@@ -154,10 +148,6 @@
 		}
 		break;
 	case 1:					/* who utmp_file */
-		if (unveil(*argv, "r") == -1)
-			err(1, "unveil");
-		if (pledge("stdio rpath getpw", NULL) == -1)
-			err(1, "pledge");
 		ufp = file(*argv);
 
 		if (only_current_term) {
@@ -183,8 +173,6 @@
 		}
 		break;
 	case 2:					/* who am i */
-		if (pledge("stdio rpath getpw", NULL) == -1)
-			err(1, "pledge");
 		ufp = file(_PATH_UTMP);
 		who_am_i(ufp);
 		break;
@@ -204,7 +192,7 @@
 	/* search through the utmp and find an entry for this tty */
 	if (mytty) {
 		while (fread((char *)&usr, sizeof(usr), 1, ufp) == 1)
-			if (*usr.ut_name && !strcmp(usr.ut_line, mytty)) {
+			if (*usr.ut_name && !strncmp(usr.ut_line, mytty, UT_LINESIZE)) {
 				output(&usr);
 				return;
 			}
@@ -215,7 +203,7 @@
 
 	pw = getpwuid(getuid());
 	(void)strncpy(usr.ut_name, pw ? pw->pw_name : "?", UT_NAMESIZE);
-	(void)time(&usr.ut_time);
+	(void)time((time_t *) &usr.ut_time);
 	*usr.ut_host = '\0';
 	output(&usr);
 }
@@ -254,7 +242,7 @@
 	}
 
 	(void)printf("%-*.*s ", UT_LINESIZE, UT_LINESIZE, up->ut_line);
-	(void)printf("%.12s ", ctime(&up->ut_time) + 4);
+	(void)printf("%.12s ", ctime((long int *) &up->ut_time) + 4);
 
 	if (show_idle) {
 		if (idle < 60) 
@@ -300,13 +288,6 @@
 		err(1, "%s", name);
 		/* NOTREACHED */
 	}
-	if (show_term || show_idle) {
-		if (pledge("stdio rpath getpw", NULL) == -1)
-			err(1, "pledge");
-	} else {
-		if (pledge("stdio getpw", NULL) == -1)
-			err(1, "pledge");
-	}
 	return(ufp);
 }
 
