--- uname.c.orig	2020-06-29 15:34:05.201755764 -0400
+++ uname.c	2020-06-29 15:38:38.132497716 -0400
@@ -57,9 +57,6 @@
 	int space = 0;
 	int print_mask = 0;
 
-	if (pledge("stdio", NULL) == -1)
-		err(1, "pledge");
-
 	while ((c = getopt(argc, argv, "amnrsvp")) != -1 ) {
 		switch (c) {
 		case 'a':
@@ -129,7 +126,7 @@
 		if (space++)
 			putchar(' ');
 
-		fputs(MACHINE_ARCH, stdout);
+		fputs(u.machine, stdout);
 	}
 	putchar('\n');
 
