--- devname.c.orig	2020-06-07 21:01:05.831174670 -0400
+++ devname.c	2020-06-07 21:03:44.228471584 -0400
@@ -41,8 +41,8 @@
 #include <string.h>
 #include <stdlib.h>
 
-static char *
-devname_nodb(dev_t dev, mode_t type)
+char *
+devname(dev_t dev, mode_t type)
 {
 	static char buf[NAME_MAX + 1];
 	char *name = NULL;
@@ -65,39 +65,3 @@
 	closedir(dirp);
 	return (name);
 }
-
-/*
- * Keys in dev.db are a mode_t followed by a dev_t.  The former is the
- * type of the file (mode & S_IFMT), the latter is the st_rdev field.
- * Note that the structure may contain padding.
- */
-char *
-devname(dev_t dev, mode_t type)
-{
-	static DB *db;
-	static bool failure;
-	struct {
-		mode_t type;
-		dev_t dev;
-	} bkey;
-	DBT data, key;
-	char *name = NULL;
-
-	if (!db && !failure) {
-		if (!(db = dbopen(_PATH_DEVDB, O_RDONLY, 0, DB_HASH, NULL)))
-			failure = true;
-	}
-	if (!failure) {
-		/* Be sure to clear any padding that may be found in bkey.  */
-		memset(&bkey, 0, sizeof(bkey));
-		bkey.dev = dev;
-		bkey.type = type;
-		key.data = &bkey;
-		key.size = sizeof(bkey);
-		if ((db->get)(db, &key, &data, 0) == 0)
-			name = data.data;
-	} else {
-		name = devname_nodb(dev, type);
-	}
-	return (name ? name : "??");
-}
