--- compat/merge.c.orig	2021-06-12 05:39:01.126680455 +0200
+++ compat/merge.c	2021-06-12 05:43:51.018012407 +0200
@@ -52,11 +52,15 @@ __FBSDID("$FreeBSD$");
  * (The default is pairwise merging.)
  */
 
+#include <sys/types.h>
 #include <sys/param.h>
 
 #include <errno.h>
 #include <stdlib.h>
 #include <string.h>
+#include <stdint.h>
+
+#include "compat.h"
 
 #ifdef I_AM_MERGESORT_B
 #include "block_abi.h"
