--- tail/tail.c.orig	2021-06-28 22:43:14.736015942 +0200
+++ tail/tail.c	2021-06-28 23:04:48.569488093 +0200
@@ -46,11 +46,9 @@ static const char copyright[] =
 static const char sccsid[] = "@(#)tail.c	8.1 (Berkeley) 6/6/93";
 #endif
 
-#include <sys/capsicum.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 
-#include <capsicum_helpers.h>
 #include <err.h>
 #include <errno.h>
 #include <getopt.h>
@@ -59,13 +57,9 @@ static const char sccsid[] = "@(#)tail.c
 #include <string.h>
 #include <unistd.h>
 
-#include <libcasper.h>
-#include <casper/cap_fileargs.h>
-
 #include "extern.h"
 
 int Fflag, fflag, qflag, rflag, rval, no_files;
-fileargs_t *fa;
 
 static file_info_t *files;
 
@@ -91,7 +85,6 @@ main(int argc, char *argv[])
 	int i, ch, first;
 	file_info_t *file;
 	char *p;
-	cap_rights_t rights;
 
 	/*
 	 * Tail's options are weird.  First, -n10 is the same as -n-10, not
@@ -162,22 +155,6 @@ main(int argc, char *argv[])
 
 	no_files = argc ? argc : 1;
 
-	cap_rights_init(&rights, CAP_FSTAT, CAP_FSTATFS, CAP_FCNTL,
-	    CAP_MMAP_R);
-	if (fflag)
-		cap_rights_set(&rights, CAP_EVENT);
-	if (caph_rights_limit(STDIN_FILENO, &rights) < 0 ||
-	    caph_limit_stderr() < 0 || caph_limit_stdout() < 0)
-		err(1, "can't limit stdio rights");
-
-	fa = fileargs_init(argc, argv, O_RDONLY, 0, &rights, FA_OPEN);
-	if (fa == NULL)
-		err(1, "unable to init casper");
-
-	caph_cache_catpages();
-	if (caph_enter_casper() < 0)
-		err(1, "unable to enter capability mode");
-
 	/*
 	 * If displaying in reverse, don't permit follow option, and convert
 	 * style values.
@@ -215,7 +192,7 @@ main(int argc, char *argv[])
 			file->file_name = strdup(fn);
 			if (! file->file_name)
 				errx(1, "Couldn't malloc space for file name.");
-			file->fp = fileargs_fopen(fa, file->file_name, "r");
+			file->fp = fopen(file->file_name, "r");
 			if (file->fp == NULL ||
 			    fstat(fileno(file->fp), &file->st)) {
 				if (file->fp != NULL) {
@@ -233,7 +210,7 @@ main(int argc, char *argv[])
 		free(files);
 	} else if (*argv) {
 		for (first = 1; (fn = *argv++);) {
-			if ((fp = fileargs_fopen(fa, fn, "r")) == NULL ||
+			if ((fp = fopen(fn, "r")) == NULL ||
 			    fstat(fileno(fp), &sb)) {
 				ierr(fn);
 				continue;
@@ -271,7 +248,6 @@ main(int argc, char *argv[])
 		else
 			forward(stdin, fn, style, off, &sb);
 	}
-	fileargs_free(fa);
 	exit(rval);
 }
 
