diff -up join/join.c.orig join/join.c
--- join/join.c.orig	2021-04-27 23:52:35.234551437 -0400
+++ join/join.c	2021-04-30 09:34:36.343074799 -0400
@@ -276,9 +276,9 @@ static void
 slurp(INPUT *F)
 {
 	LINE *lp, *lastlp, tmp;
-	size_t len;
+	size_t len = 0;
 	int cnt;
-	char *bp, *fieldp;
+	char *bp = NULL, *fieldp;
 
 	/*
 	 * Read all of the lines from an input file that have the same
@@ -321,7 +321,7 @@ slurp(INPUT *F)
 			F->pushbool = 0;
 			continue;
 		}
-		if ((bp = fgetln(F->fp, &len)) == NULL)
+		if (getline(&bp, &len, F->fp) == -1)
 			return;
 		if (lp->linealloc <= len + 1) {
 			lp->linealloc += MAX(100, len + 1 - lp->linealloc);
@@ -373,8 +373,10 @@ mbssep(char **stringp, const wchar_t *de
 		return (NULL);
 	for (tok = s;;) {
 		n = mbrtowc(&c, s, MB_LEN_MAX, NULL);
-		if (n == (size_t)-1 || n == (size_t)-2)
-			errc(1, EILSEQ, NULL);	/* XXX */
+		if (n == (size_t)-1 || n == (size_t)-2) {
+			errno = EILSEQ;
+			err(1, NULL);	/* XXX */
+		}
 		s += n;
 		spanp = delim;
 		do {
