--- csplit.c.orig	2020-06-25 09:46:14.452015278 -0400
+++ csplit.c	2020-06-25 09:56:52.460510964 -0400
@@ -100,9 +100,6 @@
 	char *ep, *p;
 	FILE *ofp;
 
-	if (pledge("stdio rpath wpath cpath", NULL) == -1)
-		err(1, "pledge");
-
 	kflag = sflag = 0;
 	prefix = "xx";
 	sufflen = 2;
@@ -237,8 +234,10 @@
 	FILE *fp;
 
 	if ((size_t)snprintf(currfile, sizeof(currfile), "%s%0*ld", prefix,
-	    (int)sufflen, nfiles) >= sizeof(currfile))
-		errc(1, ENAMETOOLONG, "%s", currfile);
+	    (int)sufflen, nfiles) >= sizeof(currfile)) {
+		errno = ENAMETOOLONG;
+		err(1, "%s", currfile);
+	}
 	if ((fp = fopen(currfile, "w+")) == NULL)
 		err(1, "%s", currfile);
 	nfiles++;
@@ -376,7 +375,7 @@
 	} else
 		ofs = 0;
 
-	if (regcomp(&cre, re, REG_BASIC|REG_NOSUB) != 0)
+	if (regcomp(&cre, re, REG_NOSUB) != 0)
 		errx(1, "%s: bad regular expression", re);
 
 	if (*expr == '/')
